server/bundler/utils.js

```diff
@@ -14,6 +14,9 @@ function getAssets( stats ) {
 }
 
 function pathToRegExp( path ) {
+	if ( path === '/' ) {
+		return path;
+	}
 	return new RegExp( '^' + path + '(/.*)?$' );
```

Comment:
IIRC all this is really doing is, given something like [this](https://github.com/Automattic/wp-calypso/blob/854c7104d13df6c99a2b33a14c93d5d379a6dadb/client/wordpress-com.js#L158), route both `/design` and `/design/*` to the corresponding module.

Both `page` and `express` have internal route parsers (that understand route fragments etc), so if we made our routes explicit in `wordpress-com.js`, I think we could drop the `RegEx` here. (Meaning we'd have to replace `path: [ '/design' ]` by `path: [ '/design', '/design/* ]`.)